arch/common/gen_isr_tables.py

```diff
@@ -209,6 +206,8 @@ def main():
     prefix = endian_prefix()
     numisrs = intlist["num_isrs"]
 
+    spurious_handler = "&_irq_spurious"
```

Comment:
This hides reference to _irq_spurious making code harder to read.
In my opinion we should put a comment near definition of _irq_spurious stating something like that:
The name of the symbol cannot be changed as it is referenced by generated code. For more information look into gen_isr_tables.py.

Reply:
I agree that this is problematic. But I have a different solution proposal.

What I am actually doing here, is trying to find/rework/create some mechanism that does
code generation according to best-practices.

When I have that I can use @b0661's codegen infrastructure here to prove that codegen will be a useful tool to support our code-generation use-cases.

With codegen it will eventually look something like this:

```
/* Not AUTO-GENERATED by gen_isr_tables.py, feel free to edit! */

#include <toolchain.h>
#include <linker/sections.h>
#include <sw_isr_table.h>
#include <arch/cpu.h>

// Referenced by irq_vector_table.h
#if defined(CONFIG_GEN_SW_ISR_TABLE) && defined(CONFIG_GEN_IRQ_VECTOR_TABLE)
#define ISR_WRAPPER ((u32_t)&_isr_wrapper)
#else
#define ISR_WRAPPER NULL
#endif

// Referenced by sw_isr_table.h
#define IRQ_SPURIOUS (&_irq_spurious)

u32_t __irq_vector_table _irq_vector_table[26] = {
    #include <generated/irq_vector_table.h>
};

struct _isr_table_entry __sw_isr_table _sw_isr_table[26] = {
    #include <generated/sw_isr_table.h>
};
```

The above code will not be generated, so the user will easily be able to understand that _irq_spurious is used by sw_isr_table.h.

Reply:
The latest version places _irq_spurious into the table as a placeholder. This will make it clearer how _irq_spuirous is used (and fix a bug).

Reply:
This indeed looks better.